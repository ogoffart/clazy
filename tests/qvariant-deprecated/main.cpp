#include <QtCore/QVariant>

int foo(const QVariant &v) {
    switch (v.type()) {
        case QVariant::Map: return 0;
        case QVariant::List: return 1;
        case QVariant::Int: return 2;
        default: break;
    }

    if (v.type() == QVariant::Line) return 3;
    return 4;
}
