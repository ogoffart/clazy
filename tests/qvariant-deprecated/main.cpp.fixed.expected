#include <QtCore/QVariant>

int foo(const QVariant &v) {
    switch (v.userType()) {
        case QMetaType::QVariantMap: return 0;
        case QMetaType::QVariantList: return 1;
        case QMetaType::Int: return 2;
        default: break;
    }

    if (v.userType() == QMetaType::QLine) return 3;
    return 4;
}
